src/fides/api/middleware.py

```diff
@@ -0,0 +1,113 @@
+import json
+from typing import Any, Dict, List
+
+from fastapi import Request
+from loguru import logger
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette.types import Message
+
+from fides.api.api import deps
+from fides.api.ctl.sql_models import AuditLogResource  # type: ignore[attr-defined]
+from fides.api.oauth.utils import extract_client_id
+
+
+async def handle_audit_log_resource(request: Request) -> None:
+    """
+    Handles the lifecycle of recording audit log resource data.
+
+    Attempts to track the WHO, WHEN, and beginning of WHAT for
+    traceability purposes.
+
+    WHO: User ID from the API request
+    WHEN: Timestamps related to the request
+    WHAT: The endpoint, request type, and (if applicable)
+    fides_key(s) associated with the request
+    """
+
+    # details to be stored as a row on the server
+    audit_log_resource_data = {
+        "user_id": None,
+        "request_path": request.scope["path"],
+        "request_type": request.method,
+        "fides_keys": None,
+        "extra_data": None,
+    }
+    db: Session = deps.get_api_session()
+
+    # get the user id associated with the request
+    token = request.headers.get("authorization")
+    if token:
+        client = await extract_client_id(token.replace("Bearer ", ""), db)
```

Comment:
looks like you'd extracted (pun intended 😉 ) this into a utility function in `get_client_user_id` and just forgot to invoke that? i like the refactor to put that in a utility function i think, but also good with not using it. but feels redundant/confusing to me as it is now!

Reply:
Yep ha, did some reorganization in test writing and missed - nice catch!